tests/api/utils/projects/test_manager.py

```diff
@@ -0,0 +1,337 @@
+async def second_nop_consumer(
+    projects_manager: mlrun.api.utils.projects.manager.ProjectsManager,
+) -> mlrun.api.utils.projects.consumers.base.Consumer:
+    return projects_manager._consumers["nop2"]
+
+
+@pytest.fixture()
+async def nop_master(
+    projects_manager: mlrun.api.utils.projects.manager.ProjectsManager,
+) -> mlrun.api.utils.projects.consumers.base.Consumer:
+    return projects_manager._master_consumer
+
+
+def test_projects_sync_consumer_project_adoption(
+    db: sqlalchemy.orm.Session,
+    projects_manager: mlrun.api.utils.projects.manager.ProjectsManager,
+    nop_consumer: mlrun.api.utils.projects.consumers.base.Consumer,
+    second_nop_consumer: mlrun.api.utils.projects.consumers.base.Consumer,
+    nop_master: mlrun.api.utils.projects.consumers.base.Consumer,
+):
+    project_name = "project-name"
+    project_description = "some description"
+    nop_consumer.create_project(
+        None,
+        mlrun.api.schemas.Project(name=project_name, description=project_description),
+    )
+    _assert_project_in_consumers([nop_consumer], project_name, project_description)
+    _assert_no_projects_in_consumers([nop_master, second_nop_consumer])
+
+    projects_manager._sync_projects()
+    _assert_project_in_consumers(
+        [nop_master, nop_consumer, second_nop_consumer],
+        project_name,
+        project_description,
+    )
+
+
+def test_projects_sync_master_project_syncing(
+    db: sqlalchemy.orm.Session,
+    projects_manager: mlrun.api.utils.projects.manager.ProjectsManager,
+    nop_consumer: mlrun.api.utils.projects.consumers.base.Consumer,
+    second_nop_consumer: mlrun.api.utils.projects.consumers.base.Consumer,
+    nop_master: mlrun.api.utils.projects.consumers.base.Consumer,
+):
+    project_name = "project-name"
+    project_description = "some description"
+    nop_master.create_project(
+        None,
+        mlrun.api.schemas.Project(name=project_name, description=project_description),
+    )
+    _assert_project_in_consumers([nop_master], project_name, project_description)
+    _assert_no_projects_in_consumers([nop_consumer, second_nop_consumer])
+
+    projects_manager._sync_projects()
+    _assert_project_in_consumers(
+        [nop_master, nop_consumer, second_nop_consumer],
+        project_name,
+        project_description,
+    )
+
+
+def test_projects_sync_multiple_consumer_project_adoption(
+    db: sqlalchemy.orm.Session,
+    projects_manager: mlrun.api.utils.projects.manager.ProjectsManager,
+    nop_consumer: mlrun.api.utils.projects.consumers.base.Consumer,
+    second_nop_consumer: mlrun.api.utils.projects.consumers.base.Consumer,
+    nop_master: mlrun.api.utils.projects.consumers.base.Consumer,
+):
+    project_name = "project-name"
+    project_description = "some description"
+    nop_consumer.create_project(
+        None,
+        mlrun.api.schemas.Project(name=project_name, description=project_description),
+    )
+    second_nop_consumer.create_project(
```

Comment:
have second_nop_consumer have a second project (different one)

and assert both are in all "consumers" + master after sync